app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -49,23 +44,35 @@ def xms_mirid_hash
       # according to fields we need to retrieve from the claim.
       # ultimately, transforming "/key1/value1/key2/value2" to {"key1" => "value1", "key2" => "value2"}
       def parsed_xms_mirid
-        split_xms_mirid = @xms_mirid_token_field.split('/')
+        begin
+          split_xms_mirid = @xms_mirid_token_field.split('/')
 
-        if split_xms_mirid.first == ''
-          split_xms_mirid = split_xms_mirid.drop(1)
-        end
+          if split_xms_mirid.first == ''
+            split_xms_mirid = split_xms_mirid.drop(1)
+          end
 
-        index = 0
-        split_xms_mirid.each_with_object({}) do |property, xms_mirid_hash|
-          case property
-          when "subscriptions"
-            xms_mirid_hash["subscriptions"] = split_xms_mirid[index + 1]
-          when "resourcegroups"
-            xms_mirid_hash["resourcegroups"] = split_xms_mirid[index + 1]
-          when "providers"
-            xms_mirid_hash["providers"] = split_xms_mirid[index + 1, index + 3]
+          index = 0
+          split_xms_mirid.each_with_object({}) do |property, xms_mirid_hash|
+            case property
+            when "subscriptions"
+              xms_mirid_hash["subscriptions"] = split_xms_mirid[index + 1]
+            when "resourcegroups"
+              xms_mirid_hash["resourcegroups"] = split_xms_mirid[index + 1]
+            when "providers"
+              xms_mirid_hash["providers"] = split_xms_mirid[index + 1, index + 3]
```

Comment:
Drive-by comment: is a default case needed here?

Reply:
Thanks for taking a look @sgnn7 ! There is actually no need for a default case as any other properties are not needed in the hash. We added this so we don't assume too much on the structure of the `xms_mirid` field and we assume only what we have to assume.

However, I do see a bug here as is needs to be `split_xms_mirid[index + 1, 3]`. The current code would have taken `index + 3` parts of `split_xms_mirid` and put them in the value for `xms_mirid_hash["providers"]`, as can be seen here - https://apidock.com/ruby/v1_9_3_125/Array/slice

@sigalsax  ^^

Reply:
@orenbm Fair enough - I just wanted to make sure that we wouldn't have exceptions when we got some unexpected case values :)

Reply:
good catch!